FirebaseVertexAI/Sample/ChatSample/Views/ErrorDetailsView.swift

```diff
@@ -23,7 +23,13 @@ extension HarmCategory: CustomStringConvertible {
     case .harassment: "Harassment"
     case .hateSpeech: "Hate speech"
     case .sexuallyExplicit: "Sexually explicit"
-    case .unknown: "Unknown"
+    case .civicIntegrity: "Civic integrity"
+    default:
+      if isUnknown() {
+        "Unknown HarmCategory: \(rawValue)"
+      } else {
+        "Unhandled HarmCategory: \(rawValue)"
+      }
```

Comment:
I understand the unknown case, but am not understanding the unhandled case. What would be a practical example of when unhandled would be encountered?

Reply:
Oh, I think I get it:
```swift
    case .dangerousContent: "Dangerous content"
    case .harassment: "Harassment"
    case .hateSpeech: "Hate speech"
    case .sexuallyExplicit: "Sexually explicit"
//    case .civicIntegrity: "Civic integrity" // if this wasn't here
    default:
      if isUnknown() {
        "Unknown HarmCategory: \(rawValue)"
      } else {
        "Unhandled HarmCategory: \(rawValue)" // then .civicIntegrity may match here because struct switching isn't exhaustive
      }
```

Reply:
Good question. Since `HarmCategory` is no longer an enum, the compiler no longer enforces that switch cases are checked exhaustively (or rather it enforces that you have a `default:` to ensure they're exhaustively checked).

Example Scenario - The developer updates to the new version of the SDK which now includes a "case" `.civicIntegrity` and they already have the following code in their app:
```
extension HarmCategory: CustomStringConvertible {
  public var description: String {
    switch self {
    case .dangerousContent: "Dangerous content"
    case .harassment: "Harassment"
    case .hateSpeech: "Hate speech"
    case .sexuallyExplicit: "Sexually explicit"
    default:
      if isUnknown() {
        "Unknown HarmCategory: \(rawValue)"
      } else {
        // civicIntegrity falls into here
        "Unhandled HarmCategory: \(rawValue)"
      }
    }
  }
}
```
The "case" `.civicIntegrity` didn't exist at the time the developer wrote the code, the compiler will never tell them it was added, so it'll fall into `default:`. It's not unknown to the SDK (the new version knows about `.civicIntegrity`) but it is unhandled in the above switch cases.